src/services/upload/metadataService.ts

```diff
@@ -26,11 +27,20 @@ const NULL_PARSED_METADATA_JSON: ParsedMetadataJSON = {
 };
 
 export async function extractMetadata(
-    receivedFile: File,
+    receivedFile: File | ElectronFile,
     fileTypeInfo: FileTypeInfo
 ) {
     let extractedMetadata: ParsedExtractedMetadata = NULL_EXTRACTED_METADATA;
     if (fileTypeInfo.fileType === FILE_TYPE.IMAGE) {
+        if (!(receivedFile instanceof File)) {
```

Comment:
I think `exifr` library can handle streams too, please check that once and if yes then we could pass a stream and avoid loading the whole file in memory

Reply:
I couldn't find any option for stream, but there is something to read exif data from a file path - should we add a function for this in Electron layer?